assets/js/Common/ServerTime.js

```diff
@@ -0,0 +1,21 @@
+const globalConfig = document.querySelector('meta[name="server-time-config"]');
+const url = globalConfig?.getAttribute('data-url') || '/api/server-time';
+let offsetMs = 0;
+
+sync();
+
+export async function sync() {
+    const requestTime = Date.now();
+    const response = await fetch(url);
+    const serverTime = await response.json();
+    const roundTripTime = Date.now() - requestTime;
+    const timeWhenServerCreatedServerTime = requestTime + roundTripTime / 2;
+    offsetMs = serverTime - timeWhenServerCreatedServerTime;
+}
+
+/**
+ * @returns {int}
+ */
+export function now() {
+    return Date.now() + offsetMs;
```

Comment:
When the client changes the clock after the server has been synced, it would need to sync again. Potentially `performance.now()` can be used to do the calculations. Or a listener to `visibilitychange` could trigger a sync when the page becomes visible.